cpp/src/arrow/columnar_format-test.cc

```diff
@@ -0,0 +1,253 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "gtest/gtest.h"
+
+#include <vector>
+
+#include "arrow/api.h"
+#include "arrow/testing/util.h"
+#include "arrow/testing/gtest_util.h"
+#include "arrow/type_traits.h"
+#include "arrow/columnar_format.h"
+
+namespace arrow {
+
+template<typename data_type, typename c_type = typename data_type::c_type>
+std::shared_ptr<typename TypeTraits<data_type>::ArrayType>
+ArrayOf(const std::vector<c_type>& values) {
```

Comment:
This is useful, could you rename it `ArrayFromVector` and add the overload to gtest_util.h?

Reply:
Are you sure you want this? We already have `ArrayFromVector`, and `ArrayOf` is the wrapper over it...